### DISCUSSION TITLE
Execution model and actors (Akka) in v2?

### DISCUSSION THREAD

**Discussion by Polpetta** [jooby-project/jooby#2819] [Q&A] [answered]
Hi! :wave: 
First of all I've to say this is a great micro-web framework, I'm very pleased to have discovered it recently.

I was playing with Jooby a little when I checked whether it was possible to leverage to an actor models like Akka, which would fit greatly in an event-driven software architecture. Looking on the official website, I've discovered there's an old Akka module, available only for the [v1 version of Jooby](https://jooby.io/v1/doc/akka/).

My questions are: is it possible to port it for the v2 too? Does it make sense with the [current execution models available](https://jooby.io/#execution-model)?

**Comment by jknack** [comment]
Hi,

 Glad you like Jooby. Guess it's possible, I didn't migrate bc no one asked until now (doubt how much people use it in 1.x)

 Would you like to submit a PR?

 If you do, you need to follow what is in the 3.x branch where basically each reactive library has his own module (rx, reactive, small rye).

**Reply by Polpetta** [reply]
Hey @jknack, I've read the @agentgt considerations in the comment below and I think it makes sense to not port it since the new license model (which I didn't knew about). It is a pity really that they changed licensing and moved away from open source :/

**Comment by agentgt** [comment, marked as answer]
I would be a little careful about including Akka anywhere in the codebase. A separate module living in its own repo is fine but it is sort of not entirely free anymore. Not really OSS.

You can google around on it. Here is one link I found on it but there is plenty more:

https://alexn.org/blog/2022/09/07/akka-is-moving-away-from-open-source/

Assuming one might be paying for Akka now I would delegate to lightbend on integrating since its their stuff now. 

I know @jknack is always very open to adding modules (he is practically a saint) to Jooby's core repo but I am not of the same opinion unless they are really useful and relevant (e.g. Jackson is fine but a different model of concurrency library that requires a different JVM language not so much).

My reasons are:

* It slows down the build
* Absolutely giant maven pom file which can slow down not just the build but the IDE .
* It makes finding stuff harder as there is more noise (e.g. greping)
* It makes it harder for contributors for core because of the above
* It adds endless dependabot noise which again annoys contributors like me (just slightly for me personally but others I know get annoyed far more) if we are subscribed

**Reply by Polpetta** [reply]
Hey @agentgt thanks for the insight regarding the new licensing terms, honestly I didn't check them while I was trying to play with Jooby and Akka. After that drift away from the open source world by Lightbend, which is a pity tbh, I don't think I want to adopt Akka anymore neither take the effort to port it since technically it's not open anymore!

Thanks for answering this question!
